### ISSUE TITLE
Errors are not being output with 3.0.0

### ISSUE THREAD

**Issue by sosuke3** [ivogabe/gulp-typescript#417]
**Expected behavior:**
gulp-typescript should output build errors and not build final concat output

**Actual behavior:**
no errors are output and concat file is output with invalid file missing from output.

**Trimmed gulpfile:**
```javascript
var tsProject = $.typescript.createProject('./Scripts/tsconfig.json');
//<snip>
gulp.task('typescript', function () {
    return gulp.src(tsFiles)
        //.pipe($.newer(jsOutputFolder + 'jt-web-app.js'))
        .pipe($.sourcemaps.init())
        .pipe($.typescript(tsProject))
        .pipe($.concat('jt-web-app.js'))
        .pipe($.sourcemaps.write('./', { sourceRoot: '/Scripts/'}))
        .pipe(gulp.dest(jsOutputFolder));
});
```
And yes I tried the newer tsProject() syntax for 3.0.0 and get the same results

**tsconfig.json**
```json
{
  "compilerOptions": {
    "noImplicitAny": false,
    "noEmitOnError": true,
    "removeComments": false,
    "sourceMap": true,
    "target": "es5"
  },
  "compileOnSave": false
}
```

**Output under 2.13.6 with typescript 1.8.10**
```
[10:27:31] Starting 'typescript'...
Scripts\app\Filter\Controllers\filterPageController.ts(46,5): error TS2304: Cannot find name 'asdadsf'.
[10:27:34] TypeScript: 1 semantic error
[10:27:34] TypeScript: emit failed
[10:27:34] Finished 'typescript' after 2.48 s
```

**Output under 2.13.6 with typescript 2.0.3**
```
[10:31:39] Starting 'typescript'...
Scripts\app\Filter\Controllers\filterPageController.ts(46,5): error TS2304: Cannot find name 'asdadsf'.
Scripts\app\Filter\Controllers\filterPageController.ts(46,5): error TS2304: Cannot find name 'asdadsf'.
[10:31:41] TypeScript: 1 semantic error
[10:31:41] TypeScript: 1 emit error
[10:31:41] TypeScript: emit failed
[10:31:41] Finished 'typescript' after 2.47 s
```

**Output under 3.0.0 with typescript 2.0.3**
```
[10:21:08] Starting 'typescript'...
[10:21:11] Finished 'typescript' after 3.16 s
```

So this is definitely not something wrong with upgrading to typescript 2.0, though I do find it interesting that I get duplicate errors with the older gulp-typescript when using typescript 2.0.

**Comment by Mr-Byte** [comment]
I've also been experiencing this same problem.

**Comment by mark9white** [comment]
.pipe(tsProject(ts.reporter.defaultReporter()))

is working for me.

**Comment by ivogabe** [comment]
Thanks for reporting, the problem was that the reporter defaulted to `.reporter.defaultReporter` instead of `.reporter.defaultReporter()`. Since the reporter is defined as an interface with two optional properties, `.reporter.defaultReporter` is assignable to it and TypeScript didn't give a warning. The test inject their own reporter, so the tests didn't catch this either. I've fixed it in c8566c78f0c39bbfefad4f2730d8a1e96f06f2aa.

**Comment by ivogabe** [comment]
The fix is released in 3.0.1.
